ocaml-lsp-server/src/workspace_symbol.ml

```diff
@@ -0,0 +1,261 @@
+open Import
+
+let to_loc (loc : Warnings.loc) : Loc.t =
+  { loc_start = loc.loc_start
+  ; loc_end = loc.loc_end
+  ; loc_ghost = loc.loc_ghost
+  }
+
+let cmt_sign (cmt_infos : Cmt_format.cmt_infos) =
+  match cmt_infos.cmt_annots with
+  | Cmt_format.Implementation { Typedtree.str_type = sign; _ }
+  | Cmt_format.Interface { Typedtree.sig_type = sign; _ }
+  | Cmt_format.Packed (sign, _) ->
+    sign
+  | _ -> []
+
+let loc_of_sig_item (signature_item : Types.signature_item) =
+  match signature_item with
+  | Types.Sig_value (_, descr, _) -> descr.val_loc
+  | Types.Sig_type (_, descr, _, _) -> descr.type_loc
+  | Types.Sig_typext (_, descr, _, _) -> descr.ext_loc
+  | Types.Sig_module (_, _, descr, _, _) -> descr.md_loc
+  | Types.Sig_modtype (_, descr, _) -> descr.mtd_loc
+  | Types.Sig_class (_, descr, _, _) -> descr.cty_loc
+  | Types.Sig_class_type (_, descr, _, _) -> descr.clty_loc
+
+let id_of_sig_item (signature_item : Types.signature_item) =
+  match signature_item with
+  | Types.Sig_value (id, _, _)
+  | Types.Sig_type (id, _, _, _)
+  | Types.Sig_typext (id, _, _, _)
+  | Types.Sig_module (id, _, _, _, _)
+  | Types.Sig_modtype (id, _, _)
+  | Types.Sig_class (id, _, _, _)
+  | Types.Sig_class_type (id, _, _, _) ->
+    id
+
+let kind_of_sig_item (signature_item : Types.signature_item) : SymbolKind.t =
+  match signature_item with
+  | Types.Sig_value _ -> Function
+  | Types.Sig_type _ -> String
+  (* | Types.Sig_typext (_, _, Types.Text_exception, _) -> Constructor *)
```

Comment:
It does make sense to extract exception declaration.

Reply:
That's covered by the line below (43):

```ocaml
Types.Sig_typext _ -> Constructor
```

`ocp-index` has an `Exception` variant, hence why they handle this case differently: https://github.com/OCamlPro/ocp-index/blob/master/libs/indexBuild.ml#L355

In our case, `SymbolKind.t` hasn't any `Exception` variant, so I thought I would always map `Types.Sig_typext` to `Constructor`. I'm definitely not sure about the mapping between the `Types.signature_item` and `SymbolKind.t` so that's for sure something we'll have to revisit at some point.